File: rb/features.rb
# KolnerAdressen SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module KolnerAdressenFeatures
  def self.make_feature(name)
    case name
    when "base"
      KolnerAdressenBaseFeature.new
    when "test"
      KolnerAdressenTestFeature.new
    else
      KolnerAdressenBaseFeature.new
    end
  end
end

File: py/features.py
# KolnerAdressen SDK feature factory

from feature.base_feature import KolnerAdressenBaseFeature
from feature.test_feature import KolnerAdressenTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KolnerAdressenBaseFeature(),
        "test": lambda: KolnerAdressenTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
